### ISSUE TITLE
Support default emacs behavior and this ideal rule

### ISSUE THREAD

**Issue by scottjad** [wasamasa/shackle#25]
I couldn't see a way to get the default emacs behavior in the sequence below. [#] denote points in time for the table below and are not keystrokes.

```
[1] C-h v argv RET [2] C-x o q [3]
```

If starting with two windows, in the table below are the number of windows at different points and which window is focused at the end. I have noted discrepancies from the default emacs behavior.

```
| configuration                 | #1 window count | #2 window count | #3 window count | #3 focused window |
|-------------------------------+-----------------+-----------------+-----------------+-------------------|
| emacs default                 |               2 |               2 |               2 |                 1 |
| :other t                      |               2 |               2 |             *1* |                 1 |
| :other t :inhibit-window-quit |               2 |               2 |               2 |               *2* |
| shackle default (nil)         |               2 |             *3* |               2 |                 1 |
```

I'm not sure the shackle defaults should change, but it would be nice if the emacs default were supported somehow.

My goal is a way to support a default rule where:
- starting and ending number of windows are the same
- new windows are only created if starting with a single window (i.e. never more than two windows unless starting with more than two)
- displayed buffers are not shown in the starting window
- displayed buffers are selected
- quitting displayed buffers returns focus to the starting window

```
| #1 window count | #1 focused window | #2 window count | #2 focused window | #3 window count | #3 focused window |
|-----------------+-------------------+-----------------+-------------------+-----------------+-------------------|
|               2 |                 1 |               2 |                 2 |               2 |                 1 |
|               1 |                 1 |               2 |                 2 |               1 |                 1 |
```

In other words, if starting with two windows, the new buffer is displayed in the non-focused window, the new buffer is selected, and when quitting the new buffer there are still two windows and the focus is in the original window.

If starting with one window, the new buffer is displayed in a new window, the new window is selected, and when quiting the new buffer the new window is closed and the focus is in the original window.

Sorry about the weak issue name.

Thanks for any help.

**Comment by wasamasa** [comment]
Assuming I interpret your tabled results correctly, I cannot reproduce that when starting out with `emacs -Q`, two windows opened and `shackle-mode` on (without customizations) `C-h v argv` opens a third window. Therefore both the first and fourth row behave the same for me.

As for the other two cases, these demonstrate two problems in Emacs.  The first one would be window quitting.  I do sort of regret having introduced options and actions for altering the `quit-window` parameter of a window, because although it was easy to do it this way, it is wrong in that it's mixing up Shackle's actual purpose (managing `display-buffer` behaviour) with an unrelated topic.  Using `:inhibit-window-quit` enables this option and unsets that parameter.

The other problem seems to be deeply rooted in Emacs itself.  @Fuco1 provided `C-x 3 C-x o C-x o C-x 2 C-x 0` for reproduction.  You'd expect point to move to the parent window of the child window you've killed, but it switches to the other window instead.

**Comment by scottjad** [comment]
I should have mentioned that the results above are for C-x 3 (not 2, that's the important part) C-h v argv RET C-x o q. I think with that you'll be able to reproduce them in emacs -Q.

**Comment by wasamasa** [comment]
I can't reproduce it with this either. As I've already explained the other behaviour, I'm closing this issue.
